src/TransformerExtender.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\fb_instant_articles\TransformerExtender.
+ */
+
+namespace Drupal\fb_instant_articles;
+
+use Facebook\InstantArticles\Transformer\Transformer;
+
+/**
+ * Encapsulates Drupal-specific logic when using the Transformer class.
+ *
+ * Class TransformerExtender
+ * @package Drupal\fb_instant_articles
+ */
+class TransformerExtender extends Transformer {
+
+  /**
+   * Transformer constructor.
+   *
+   * Adds Drupal hook to allow transformer rules to be added by other modules.
+   * We invoke the hook in a constructor - as opposed to inside transform() or a
+   * new invokeRules() method - because of the way Transformer is normally used.
+   *
+   * @see hook_fb_instant_articles_transformer_rules()
+   *
+   * Note the parent class does not have a constructor, so we do not call
+   * @code parent::__construct() @endcode.
+   */
+  public function __construct() {
+    $rules = module_invoke_all('fb_instant_articles_transformer_rules');
+    $this->addRules($rules);
```

Comment:
As discussed with @scottrigby, let's add an drupal_alter to allow modules to alter the final list of rules across all implementations of the info hook.